CONTRIBUTING.md

```diff
@@ -0,0 +1,103 @@
+# Contributing
+
+When contributing to this repository, please first discuss the change you wish to make via issue, email, or any other method with the owners of this repository before making a change.
+
+## Pull Request Process
+
+The following outlines the general rules we follow:
+
+1. All PRs should have the appropriate testcases when possible. For example, bug fixes could include tests that demonstrates the issue w/o your fix.
+2. All PRs must have the appropriate documentation about changes made within the same PR. Note, not all PRs will necessarily require a documentation change but if it does please include it in the same PR so the PR is complete and not just a partial solution.
+3. All PRs must have appropriate documentation. New features should be described, and an example of use provided when relevant.
+4. All PRs should follow Code Guidelines
+
+## Coding Guidelines
+
+While code stylistics (naming, whitespace, structure) is mostly handled by the famous PEP8 - https://www.python.org/dev/peps/pep-0008/
```

Comment:
I think we should remove the rest of this document because it's duplicate to the publicly available https://leapp.readthedocs.io/en/latest/python-coding-guidelines.html. I would just link it here:
_"The code **should** follow the [Leapp Framework Coding Guidelines](https://leapp.readthedocs.io/en/latest/python-coding-guidelines.html)."_
I used "should" as those guidelines actually look more like best practices.
 
But, apart from linking the above guidelines, I'm missing the following here:
- naming convention:
  - folders and python files named in lowercase without underscores, python classes in CamelCase,
  - do not use “actor” in the actor's folder name & do not use "model/topic/tag/phase" in the model/topic/tag/phase file name, these things are implicit.)
- write code for Python 2.7+/3.6+
- the actor's main file must be named "actor.py", the library files (within the actor's _libraries_ subfolder) can be named arbitrarily
- use of linters (PyLint, pep8, flake8) is encouraged to be sure to follow PEP8
- line length up to 120 character

Reply:
I've moved the project-wide contributing guidelines to https://github.com/oamg/leapp-guidelines/contributing-guidelines.rst under https://github.com/oamg/leapp-guidelines/pull/3.
So you can link it here like this:
_"The code **should** follow the [Leapp project Python coding guidelines](https://github.com/oamg/leapp-guidelines/contributing-guidelines.rst)."_